src/solidity_to_ast4gen.coffee

```diff
@@ -201,45 +206,61 @@ walk = (root, ctx)->
       for node in root.nodes
         ret.list.push walk node, ctx
       [ret.pos, ret.line] = parse_line_pos(root.src)
+      ret.file = ctx.file[0]
       ret
     
     when "ContractDefinition"
-      ret = new ast.Class_decl
-      
-      switch root.contractKind
-        when "contract"
-          ret.is_contract = true
-        
-        when "library"
-          ret.is_library = true
-        
-        when "interface"
-          ret.is_interface = true
+      # HACK to know in which file following nodes are located in
+      if root.name.startsWith "ImportPlaceholderStart"
+        ctx.file.unshift root.nodes[0].value.value 
+        ret = new ast.Comment
+        ret.text = "here #{ctx.file[0]} was imported"
+        ret.can_skip = true
```

Comment:
can_skip is not defined in ast definition
can_skip will be lost on clone (clone can occur in ast transform)